### DISCUSSION TITLE
Possible to use your own image in a ToggleButton?

### DISCUSSION THREAD

**Discussion by josdemmers** [MahApps/MahApps.Metro#4267] [Q&A]
I was trying out the toggle button and was wondering if you can use your own images or only those from the iconPacks?

```
<ToggleButton Width="48" Height="48" Margin="4" Foreground="{DynamicResource MahApps.Brushes.Accent}" Style="{DynamicResource MahApps.Styles.ToggleButton.Circle}">
        <ToggleButton.ContentTemplate>
          <DataTemplate>
            <!--<iconPacks:PackIconModern Width="20" Height="20" Kind="City" />-->
            <Image Source="/Images/arcana_250px.png"/>
          </DataTemplate>
        </ToggleButton.ContentTemplate>
</ToggleButton>
```

The xaml view shows the image just fine but when launching the application I only have an empty toggle button.

**Comment by josdemmers** [comment]
Seems to work fine when I use a binding for the image source. No idea why it doesn't directly work in the xaml.

**Comment by fredericoregateiro** [comment]
The way i do it:
The image must be in the debug folder
```xml
<ToggleButton Width="48" Height="48" Margin="4" Foreground="{DynamicResource MahApps.Brushes.Accent}" Style="{DynamicResource MahApps.Styles.ToggleButton.Circle}">
    <Image Source="/Images/arcana_250px.png"/>
</ToggleButton>
```

or if the image is in the solution, set the build action to `Resource`
```xml
<ToggleButton Width="48" Height="48" Margin="4" Foreground="{DynamicResource MahApps.Brushes.Accent}" Style="{DynamicResource MahApps.Styles.ToggleButton.Circle}">
    <Image Source="/projectname;component/Images/arcana_250px.png"/>
</ToggleButton>
```
replace the `projectname` by the name of the project

**Reply by josdemmers** [reply]
Strange tried that one as well without success. The odd things is it works fine like that in the xaml preview of Visual Studio.
However when I start the application the icons are gone.

But it works fine now using bindings, so I'll use that for now.

**Reply by punker76** [reply]
Sometimes it's necessary to use `pack://application:,,,/projectname;component/...` to get that working.
